deeplab/README.md

```diff
@@ -0,0 +1,146 @@
+
+The new height of the processed image.
+
+* **SegmentationMap** :: Uint8ClampedArray
+
+The colored segmentation map as Uint8ClampedArray which can be [fed](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas) into ImageData and mapped to a canvas.
+
+#### Segmentation example
+
+```typescript
+const classify = async (image) => {
+    return await model.predict(image);
+}
+```
+
+**Note**: *For more granular control, consider `segment` and `translate` methods described below.*
+
+### Producing a Semantic Segmentation Map
+
+To segment an arbitrary image and generate a two-dimensional tensor with class labels assigned to each cell of the grid overlayed on the image (with the maximum number of cells on the side fixed to 513), use the `segment` method of the `SemanticSegmentation` object.
+
+#### `model.segment` input
+
+* **image** :: DeepLabInput | ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | tf.Tensor3D;
+
+#### `model.segment` output
+
+* **rawSegmentationMap** :: Promise<RawSegmentationMap | tf.Tensor2D>
+
+#### `model.segment` example
+
+```javascript
+const getSemanticSegmentationMap = async (image) => {
+    return await model.segment(image)
+}
+```
+
+### Translating a Segmentation Map into the Color-Labelled Image
+
+To transform the segmentation map into a coloured image, use the `translate` method of the `SemanticSegmentation` object. Pass an optional canvas element as `canvas` to draw the output as a side effect.
+
+#### `model.translate` input
+
+* **segmentationMap** :: RawSegmentationMap | tf.Tensor2D
+* **canvas** (optional) :: HTMLCanvasElement
+
+#### `model.translate` output
+
+The returned promise is the future of an array typed as `SegmentationData` that contains two objects:
+
+* **Legend** :: { [name: string]: [number, number, number] }
+
+The legend is a dictionary of objects recognized in the image and their colors.
+
+* **SegmentationMap** :: Uint8ClampedArray
+
+The colored segmentation map as Uint8ClampedArray which can be [fed](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas) into ImageData and mapped to a canvas.
+
+#### `model.translate` example
+
+```javascript
+const translateSegmentationMap = async (segmentationMap) => {
+    return await model.translate(segmentationMap)
+}
+```
+
+## Contributing to the Demo
+
+Please see the demo [documentation](./demo/README.md).
+
+## Technical Details
+
+This model is based on the TensorFlow [implementation](https://github.com/tensorflow/models/tree/master/research/deeplab) of DeepLab v3. You might want to inspect the [conversion script](./convert_deeplab.sh), or download original pre-trained weights [here](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md). To convert the weights locally, run the script as follows, replacing `dist` with the target directory:
+
+```bash
+./utils/convert_deeplab.sh dist
```

Comment:
Since this is more of a script, let's call this folder `scripts` and make the change here as well